### DISCUSSION TITLE
[Question] License upgrade

### DISCUSSION THREAD

**Discussion by pauloswear** [dashingsoft/pyarmor#1034] [Q&A] [answered]
Hi @jondy. If I decide upgrade my license to pyarmor 8, can I still use pyarmor 7.7.4 version on other machines?
Thanks

**Comment by jondy** [comment, marked as answer]
Yes.

**Comment by pauloswear** [comment]
If I decide upgrade, can I still generate spp code using version 7.7.4 on other machines without any change on capsule?

Image asking upgrade:
https://prnt.sc/AjShSgXd_7Sz

And about:

> Registering Pyarmor by pyarmor register pyarmor-regcode-xxxxxx.txt can be used no more than 10 times

Can I register same license more than 10 times using another way?

**Reply by jondy** [reply]
spp mode could not be used by version 7.7.4, license server doesn't serve api request from 7.7.4.

spp mode still could be used by version 8.x+ with command `pyarmor-7`.

> Can I register same license more than 10 times using another way?

Yes. There is no limitations for `.zip`.

**Comment by pauloswear** [comment]
@jondy  even using basic license can I use spp using pyarmor-7 command?

Other question, my runtime files is already generated using 7.7.4 ver, if I upgrade, can I still generate scripts with --no-runtime arg without generate new runtime ? (pyarmor 8)

**Comment by jondy** [comment]
pyarmor-7 is same as command `pyarmor` in Pyarmor 7.7.4

And you can try it in one minute instead waiting for my answer.

**Reply by pauloswear** [reply]
The major problem will be try, and lost the connection (could not upgrade anymore) with old scripts that is using --no-runtime. 
That's the reason of a lot questions

**Comment by jondy** [comment]
> Other question, my runtime files is already generated using 7.7.4 ver, if I upgrade, can I still generate scripts with --no-runtime arg without generate new runtime ? (pyarmor 8)

No.

Let's make some concepts clear.

Pyarmor 8 means 3 sub commands: `gen`, `reg`, `cfg`. If generate scripts by `gen`, pyarmor 7.7.4 runtime doesn't work.

command `pyarmor-7` even in Pyarmor 8.0.21 is not Pyarmor 8.
command `pyarmor-7` could be taken as Pyarmor 7.7.5, a bug fixed version of Pyarmor 7.7.4

If generate scripts by `pyarmor-7 obfuscate`, pyarmor 7.7.4 runtime still works.
